lib/utils/useProvider/index.tsx

```diff
@@ -199,15 +205,16 @@ export const useProvider = (): Web3ModalState => {
 
     /** There is a bug where ethers doesn't respond to web3modal events for these two, so we use the nested provider
      * https://github.com/ethers-io/ethers.js/issues/2988 */
-    web3state.provider.provider.on("accountsChanged", handleAccountsChanged);
+    web3state.provider.provider.on("accountsChanged", (accts) =>
+      handleAccountsChanged(accts)
+    );
     web3state.provider.provider.on("chainChanged", handleChainChanged);
     /** For WalletConnect and Coinbase disconnections */
     web3state.provider.provider.on("disconnect", handleDisconnect);
 
     return () => {
-      web3state.provider.provider.removeListener(
-        "accountsChanged",
-        handleAccountsChanged
+      web3state.provider.provider.removeListener("accountsChanged", (accts) =>
+        handleAccountsChanged(accts)
```

Comment:
T inline arrow function creates a new instance, so the `removeListener` might not be able to find and remove the handler that was instantiated earlier.

That's why I reuse the same reference to `handleAccountsChanged` directly

Reply:
Yeah that makes sense. Typescript was giving me trouble here hence the change to it - will take another look.